R/user-access.R

```diff
@@ -0,0 +1,39 @@
+#' Get information about when users last accessed a dataset
+#'
+#' Useful to get dataset-user interaction metadata about all of the users that
+#' have access to a particular dataset.
+#'
+#' @param dataset_entity a dataset object or a dataset entity (e.g. a single
+#' entry from `selectDatasetCatalog()`)
+#' @param user_fields metadata fields about users to include in the dataframe
+#' (default: c("name", "email", "access_time"))
+#'
+#' @return a dataframe with information about all users for a particular dataset
+#' @export
+userInfo <- function (dataset_entity, user_fields = c("name", "email", "access_time")) {
```

Comment:
I think these function names are fine, but another option is to have a single `userInfo` method which  would return different things depending on whether a dataset or project were passed to it. This might be a good way to go because we might want to add equivalent functions for other objects in the future.

Reply:
Agree with this sentiment, and it should be called `users()`. This is in keeping with our current/upcoming dataset/project API changes, which will make them more consistent.